MediaBrowser.MediaEncoding/Encoder/MediaEncoder.cs

```diff
@@ -32,323 +28,288 @@ namespace MediaBrowser.MediaEncoding.Encoder
     public class MediaEncoder : IMediaEncoder, IDisposable
     {
         /// <summary>
-        /// The _logger
-        /// </summary>
-        private readonly ILogger _logger;
-
-        /// <summary>
-        /// Gets the json serializer.
+        /// Gets the encoder path.
         /// </summary>
-        /// <value>The json serializer.</value>
-        private readonly IJsonSerializer _jsonSerializer;
+        /// <value>The encoder path.</value>
+        public string EncoderPath => FFmpegPath;
 
         /// <summary>
-        /// The _thumbnail resource pool
+        /// External: path supplied via command line
+        /// Custom: coming from UI or config/encoding.xml file
+        /// System: FFmpeg found in system $PATH
+        /// null: No FFmpeg found
         /// </summary>
-        private readonly SemaphoreSlim _thumbnailResourcePool = new SemaphoreSlim(1, 1);
-
-        public string FFMpegPath { get; private set; }
-
-        public string FFProbePath { get; private set; }
+        public string EncoderLocationType { get; private set; }
```

Comment:
this should be `enum`, not `string`

Reply:
Ideally, yes.  But believe my proposed change in jellyfin-web removes the single use of this vairable:
"External" == systemInfo.EncoderLocationType()

I'm wondering if we need it at all now...

Reply:
Well, all in all I think it might be beneficial to show it in a plaque of sorts (beware - MAD PAINT SKILLZ!)
![plaque](https://user-images.githubusercontent.com/1011526/52492317-f8962d80-2bd9-11e9-86db-318b78e75718.png)

But for showing that it should probably be localized, so it should better be a `enum`. Plus being a `enum` shows all possible values for a variable.

Reply:
ok, lets keep it and swap to enum.